### ISSUE TITLE
The training and testing data used in BPfold

### ISSUE THREAD

**Issue by Dyyyg** [heqin-zhu/BPfold#7]
Hello, I would like to ask a question that may be due to my misunderstanding. Did BPfold actually use the 652 structures from the PDB during training? The main text states that the PDB dataset is used for testing and is divided into three subsets: TS1 (60 sequences), TS2 (38 sequences), and TS3 (18 sequences). However, Table S1 describes that 652 PDB structures were used for training. Meanwhile, the paper also mentions that the training sets are RNAStrAlign and the TR0, VL0 subsets from bpRNA-1m. Could you please clarify whether these descriptions are contradictory, or if I have missed something ? Thank you for your time and guidance.

**Comment by heqin-zhu** [comment]
Hello,

PDB dataset has two parts: the 652 PDB structures are used for training and TS1/TS2/TS3 are used for testing. In the paper, following previous studies, the training datasets are composed of RNAStrAlign and bpRNA, while PDB 669 (652 structures with length no more than 600 nt) structures are included in the bpRNA dataset.

**Comment by Dyyyg** [comment]
Hello,

Thank you for your patient explanation regarding the dataset split. So, the training set consists of RNAStrAlign (19,313 sequences), bpRNA (12,114 sequences), and PDB (652 sequences), **totaling 32,079 sequences** – is that correct?

At the same time, I would like to take this opportunity to ask a few questions about the training implementation of BPfold, hoping to better understand your experiments:

**Hardware:** What GPU model(s) and how many GPUs were used for training BPfold?

**Training hyperparameters:** What are the specific hyperparameter settings for training, such as number of epochs, batch size, gradient clip threshold, and learning rate,?

**Cross‑validation:** The paper mentions using 5‑fold cross‑validation with early stopping, but I noticed that the pre-trained weights provided correspond to a 6‑fold cross‑validation setup. Does the paper use 5‑fold, but suggest using 6‑fold for inference?

Thank you very much for your time and guidance. I look forward to your reply.

Best regards！

**Comment by heqin-zhu** [comment]
Hello,

The training dataset comprises RNAStrAlign and bpRNA-1m (a mixed database, including the PDB training).

We used A10 GPU for training.

For hyperparameter details, please refer to the paper and the default values in the scripts.

We adopted 5-fold cross-validation to ensure a fair comparison with SPOT-RNA, which uses the same 5-fold setup. For the final production model (intended for real-world use rather than benchmark comparison), more folds theoretically yield better generalization performance.

**Comment by Dyyyg** [comment]
Hello,

Thank you for your reply. I checked RNAStrAlign and bpRNA-1m. I found that the PDB training dataset is not included in RNAStrAlign and bpRNA-1m; rather, the PDB training dataset is listed separately.

<img width="891" height="464" alt="Image" src="https://github.com/user-attachments/assets/e3ab258f-6ca3-454f-bf6b-63e755005c10" />

**Comment by heqin-zhu** [comment]
Please refer to the official bpRNA-1m database.
